### PR TITLE
NO-JIRA: deps: Bump to K8s 1.29.2

### PR DESCRIPTION
**Pull request by swatisehgal**
- deps: go mod updates: (Bump to K8s 1.29.2, Bump debug-tools to v0.1.12, Consume cpuset from k/utils)
- Vendor updates
- Ensure that `make vet` passes 
- drop deprecated pointer package

### CODE REVIEW

**Comment by swatisehgal** [comment]
/hold
Placeholder PR. @jmencak Let me know if we prefer to bump to the latest version i.e. 1.29.2 and I will update.

**Comment by swatisehgal** [comment]
Ah, sorry intended to add as reviewer!

/unassign @jmencak 
/cc @jmencak

**Comment by jmencak** [comment]
Thank you for the PR, Swati.  Why are we bumping to 1.29.0 since we already have [1.29.2](https://git.k8s.io/kubernetes/CHANGELOG/CHANGELOG-1.29.md#v1292)?  Also, `make vet` needs to pass first.

**Comment by swatisehgal** [comment]
Depends on https://github.com/openshift-kni/debug-tools/pull/82.

File: go.mod
Comment (swatisehgal) [inline go.mod:241]: I am having trouble bumping to latest version of MCO so for now updated it to a more upto date version of release-4.14 while I am figuring out bump to 4.16.

Comment (hexfusion) [reply go.mod:241]: Yeah I wonder if we need to first bump openshift/api openshift/client-go first removing MCO as a dep entirely.

Comment (swatisehgal) [reply go.mod:241]: Are you suggesting that I remove commit that updates MCO deps from this PR?
Removing MCO dependency entirely doesn't seem right to me...

Comment (hexfusion) [reply go.mod:241]: From what I can tell the MCO dependency in this repo will be replaced by openshift/api and openshift/client-go because your only dependant on those types which the new MCO doesn't export anymore. So my question is does that update ned to happen before k8s. I am going to test it in another PR on top of this now.

Comment (hexfusion) [reply go.mod:241]: https://github.com/openshift/cluster-node-tuning-operator/pull/1021 this is built on your PR

Comment (swatisehgal) [reply go.mod:241]: Ah, I understand what you meant now :)

Comment (hexfusion) [reply go.mod:241]: I got to the root of the failure this file is manually maintained and I think we just need to update the new API field

https://github.com/openshift/cluster-node-tuning-operator/pull/1021/commits/23ff3edb5be20c72a417ab1cefec2c872ac81b70

Comment (hexfusion) [reply go.mod:241]: passes now just needed to update everywhere https://github.com/openshift/cluster-node-tuning-operator/pull/1021

Comment (swatisehgal) [reply go.mod:241]: Thanks for the deep dive! Updating my now.

**Comment by swatisehgal** [comment]
Moved the OCP deps update and MCO deps removal to a separate PR: https://github.com/openshift/cluster-node-tuning-operator/pull/1022. Once this PR is merged, will rebase that.

Thanks @hexfusion for helping out here!

File: pkg/operator/profilecalculator.go
Comment (jmencak) [inline pkg/operator/profilecalculator.go:246]: This change is not correct, but it highlights an issue in the code. %s/%s is intended, but `recommendAll[iStop].Profile` and `recommendAll[i].Profile` are pointers to a string. They are both required values, so dereferencing them shouldn't result in a crash. But the Priority is also a required value.  I'd say we either we want to keep %s/%s and dereference recommendAll[iStop].Profile and recommendAll[i].Profile or remove the if clause completely, because Priority is also required.

File: pkg/operator/profilecalculator.go
Comment (jmencak) [inline pkg/operator/profilecalculator.go:350]: Same as above

**Review by jmencak — CHANGES_REQUESTED** [review]
Thank you for the PR. Looks good overall, just noticed an issue in the profilecalculator code that needs to be addressed.

As for the correctness of the PPC changes, deferring to the CNF team for review.

File: pkg/operator/profilecalculator.go
Comment (yanirq) [reply pkg/operator/profilecalculator.go:246]: If the change is not directly related to the bump then we can do it in another PR.

Comment (Tal-or) [reply pkg/operator/profilecalculator.go:246]: I think it is, since it stems from the klog bump

Comment (swatisehgal) [reply pkg/operator/profilecalculator.go:246]: Thanks for pointing this out, will fix by dereferencing the pointers.

**Comment by jmencak** [comment]
Thank you for the changes.
/approve

**Comment by sinnykumari** [comment]
Thanks @swatisehgal for working on the kube update. Should we removed hold if this is ready to get merged?

**Comment by swatisehgal** [comment]
> Thanks @swatisehgal for working on the kube update. Should we removed hold if this is ready to get merged?

Yes, Thanks for pointing that out.
/unhold

**Comment by yanirq** [comment]
/retitle NO-JIRA: deps: Bump to K8s 1.29.2

**Comment by Tal-or** [comment]
/retest
```
2024-04-09T05:43:17Z	ERROR	Failed to create cluster	{"error": "failed to create infra: cannot list vpcs: UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:iam::820196288204:user/hypershift-ci-jobs is not authorized to perform: ec2:DescribeVpcs with an explicit deny in a service control policy\n\tstatus code: 403, request id: 837d4c69-9a00-4130-ab09-10d70a61e6c3"}
github.com/openshift/hypershift/cmd/cluster/aws.NewCreateCommand.func1
	/hypershift/cmd/cluster/aws/create.go:71
github.com/spf13/cobra.(*Command).execute
	/hypershift/vendor/github.com/spf13/cobra/command.go:983
github.com/spf13/cobra.(*Command).ExecuteC
	/hypershift/vendor/github.com/spf13/cobra/command.go:1115
github.com/spf13/cobra.(*Command).Execute
	/hypershift/vendor/github.com/spf13/cobra/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext
	/hypershift/vendor/github.com/spf13/cobra/command.go:1032
main.main
	/hypershift/main.go:78
runtime.main
	/usr/lib/golang/src/runtime/proc.go:267
Error: failed to create infra: cannot list vpcs: UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:iam::820196288204:user/hypershift-ci-jobs is not authorized to perform: ec2:DescribeVpcs with an explicit deny in a service control policy
	status code: 403, request id: 837d4c69-9a00-4130-ab09-10d70a61e6c3
failed to create infra: cannot list vpcs: UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:iam::820196288204:user/hypershift-ci-jobs is not authorized to perform: ec2:DescribeVpcs with an explicit deny in a service control policy
	status code: 403, request id: 837d4c69-9a00-4130-ab09-10d70a61e6c3
```
Both hypershift lanes suffer from the same issue

**Comment by swatisehgal** [comment]
/override ci/prow/e2e-hypershift-pao
/override ci/prow/e2e-hypershift

Hypershift lane failure is a known issue and is currently being investigated.

**Pull request merged**
